### PR TITLE
generate 3.0 operators

### LINKED ISSUE #1310: MongoDB 3.0: use new operators
Depends on #1308

There is only one new operator: [`$dateToString`](https://docs.mongodb.com/manual/reference/operator/aggregation/dateToString/#exp._S_dateToString).
- [x] Use it to implement `TimeOfDay` in expr instead of Js (avoiding mapreduce).
- [ ] Possibly add a new date formatting function (possibly following the [PostgreSQL syntax](https://www.postgresql.org/docs/9.2/static/functions-formatting.html)).

Note: selectors now support [`$eq`](https://docs.mongodb.com/manual/reference/operator/query/eq/#op._S_eq), but the benefit of that for generated worklfows is probably not worth the effort to track types in `Selector`.

### PR DESCRIPTION
**Pull request by mossprescott**
Closes #1310.

The upshot here is just that `time_of_day` can now be evaluated in the aggregation pipeline. More interestingly, this lays the groundwork for the 3.2 ops to be generated as well.

This is based on #1434, which should be merged by the time this is ready to go, which makes the diff here a little bit useless. Once that's merged, I'll review and see if it makes more sense to resubmit with a cleaner history.

### CODE REVIEW

**Comment by alissapajer** [comment]
MountServiceSpec:
```
[error]       x succeed with view path (45 seconds, 858 ms)
[error]        Falsified after 89 passed tests.
[error]        > ARG_0: FileIn(DirIn(DirIn(DirIn(DirIn(DirIn(DirIn(DirIn(Current,DirName(/tĮnrenh/3cp)),DirName(c8.eq/f)),DirName(r틳ovo)),DirName(䡅)),DirName(blwz)),DirName(jeox6jdtmfm.cx)),DirName(3)),FileName(H))
[error]        > ARG_1: DirIn(DirIn(DirIn(DirIn(DirIn(Root,DirName(i/1MbnO.c2j//fowubfhowqek./.jk/ka.Ov.5g.9z+drdYohqd.if/y/gr..6f..i.v/.)),DirName(g뱿r5lou)),DirName(eXy)),DirName(t)),DirName(j))
[error]        > 'added /i$sep$1MbnO.c2j$sep$$sep$fowubfhowqek.$sep$.jk$sep$ka.Ov.5g.9z drdYohqd.if$sep$y$sep$gr..6f..i.v$sep$./g뱿r5lou/eXy/t/j/$sep$tĮnrenh$sep$3cp/c8.eq$sep$f/r틳ovo/䡅/blwz/jeox6jdtmfm.cx/3/H'
[error]        
[error]         is not equal to 
[error]        
[error]        'added /i$sep$1MbnO.c2j$sep$$sep$fowubfhowqek.$sep$.jk$sep$ka.Ov.5g.9z+drdYohqd.if$sep$y$sep$gr..6f..i.v$sep$./g뱿r5lou/eXy/t/j/$sep$tĮnrenh$sep$3cp/c8.eq$sep$f/r틳ovo/䡅/blwz/jeox6jdtmfm.cx/3/H' (QuasarSpecification.scala:31)
[error] Actual:   ...5g.9z[ ]drdYo...
[error] Expected: ...5g.9z[+]drdYo...
```

**Comment by mossprescott** [comment]
This commit history is just too hard to follow, so I'm going to close this, rebase, and resubmit.
